File: alibaba-java-coding-guidelines-test-master/app/src/main/java/com/itlong/detect/code/alibabajavacodingguidelinestest/lsn01/CommonUtil.java
package com.itlong.detect.code.alibabajavacodingguidelinestest.lsn01;

import android.content.Context;

/**
 * @author HuQingHui
 */
public class CommonUtil {

    private static CommonUtil instance;
    private Context context;

    private CommonUtil(Context context) {
        this.context = context;
    }

    public static CommonUtil getInstance(Context context){
        if(instance == null){
            instance = new CommonUtil(context);
        }
        return instance;
    }
}

File: DetektTest/app/src/main/java/com/lovejjfg/detekttest/lsn01/CommonUtil.kt
package com.lovejjfg.detekttest.lsn01

import android.content.Context

/**
 * 描述：
 * 公司：深圳市旺龙智能科技有限公司
 * @author HuQingHui
 * @date 2022/9/1
 */
class CommonUtil {
    private var context: Context? = null

    constructor(context: Context) {
        this.context = context
    }

    companion object {
        private var instance: CommonUtil? = null
        @JvmStatic
        fun getInstance(context: Context): CommonUtil? {
            if (instance == null) {
                instance = CommonUtil(context)
            }
            return instance
        }
    }
}
